### PR TITLE
Hello world revision

### PR DESCRIPTION
**Pull request by nostophilia**
Demonstration of #579

Revise some part of the hello world example with Redux

---

This change is [Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails/584)

### CODE REVIEW

**Comment by justin808** [comment]
@robwise @jbhatab @Judahmeek I'd like to get your feedback on this one. I'm going to create a sample repo with a PR generated from the generator.

**Comment by justin808** [comment]
@robwise @jbhatab @Judahmeek @nostophilia I created: https://github.com/shakacode/react_on_rails-test-new-redux-generation/pull/1

Let's comment on the code there and here.

**Comment by justin808** [comment]
@nostophilia 

1. Please see: https://github.com/shakacode/react_on_rails-test-new-redux-generation/pull/2. We need to generate code that passes linting.
2. We can cleanup the comments. We should remove the revision dates from the generated files. The more doc references we provide the better.

Overall, I think this looks great. Just needs a bit of cleanup!

Thanks for your contributions. 

I'd like to get this merged in the next day or so. If you cannot get to it, please let me know. I'd prefer to have you make the changes so that your name is in the commit history with one big commit.

**Comment by justin808** [comment]
@nostophilia Please let me know when you're ready. Please confirm if you got all the requested changes and if linting passes using the linter 13.0.0.

`npm i --save-dev eslint-config-shakacode@13.0.0` after running the generator.

**Comment by nostophilia** [comment]
@justin808 
It's ready except this [violation](https://github.com/shakacode/react_on_rails-test-new-redux-generation/pull/5#pullrequestreview-7265643)

**Comment by justin808** [comment]
https://github.com/shakacode/react_on_rails-test-new-redux-generation/pull/8

So close! Just need to fix the lint errors on what's generated.

BE SURE to use the linters in this commit: 

https://github.com/shakacode/react_on_rails-test-new-redux-generation/pull/8/commits/ef1bde4480680e1805e59c6b09becad62b99e614

You can cherry-pick this!

File: lib/generators/react_on_rails/templates/no_redux/base/client/app/bundles/HelloWorld/containers/HelloWorldContainer.jsx [hunk 1]
```diff
@@ -15,14 +15,12 @@ export default class HelloWorld extends React.Component {
...
-  updateName(name) {
-    this.setState({ name });
-  }
+  updateName = (name) => { this.setState({ name }); };
 
   render() {
     return (
       <div>
-        <HelloWorldWidget name={this.state.name} updateName={e => this.updateName(e)} />
+        <HelloWorld name={this.state.name} updateName={this.updateName} />
```
Comment (nostophilia) [inline lib/generators/react_on_rails/templates/no_redux/base/client/app/bundles/HelloWorld/containers/HelloWorldContainer.jsx]: Please note that there will be few changes in no-redux example as HelloWorldWidget.jsx is shared. Also, there's a lint violation here. And I simplified it a little bit with the [arrow function](https://babeljs.io/blog/2015/06/07/react-on-es6-plus#arrow-functions) way.

Comment (mongjong59) [inline lib/generators/react_on_rails/templates/no_redux/base/client/app/bundles/HelloWorld/containers/HelloWorldContainer.jsx:23]: Please note that there will be a few small changes related to file and component name in no-redux example as HelloWorldWidget.jsx is shared. 

Also, there's a lint violation here. And I simplified it a little bit with the [arrow function](https://babeljs.io/blog/2015/06/07/react-on-es6-plus#arrow-functions).

**Comment by justin808** [comment]
@nostophilia when will you push v6 to https://github.com/shakacode/react_on_rails-test-new-redux-generation/ ? I'd like to get this change into the next release coming maybe this weekend.

**Comment by justin808** [comment]
:lgtm_strong: 

**AMAZING JOB!** Great teamwork! Big improvement!

I'm shipping this!

---

Reviewed 18 of 19 files at r1, 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/584#-:-KWyGyAOATiHosXSdpto:b-uqrjot)*

**Pull request merged**